### ISSUE TITLE
Pull out core functionality into new repo

### ISSUE THREAD

**Issue by delphidabbler** [delphidabbler/vilib#8]
The core functionality may be useful to other projects ([VIEd](https://github.com/delphidabbler/vied), for example).

So why not extract that code (I.e. `TVerInfoData` and all the classes it depends upon) and move into a library repo).

That would leave behind the project file, the interface definitions, the interface implementation in `TVerInfoBinary` and any demo code (see issue #6)

The core functionality could be imported via a submodule or similar.

**Note** This proposal is dependent upon issue #7 being implemented first.

**Comment by delphidabbler** [comment]
Such a new library could also include the version information stream listed in the documentation.

**Comment by delphidabbler** [comment]
> Such a new library could also include the version information stream listed in the documentation.

In fact, methods to read/write from/to Delphi streams could be provided in `TVerInfoData` to complement the `IStream` methods currently provided.

**Comment by delphidabbler** [comment]
Awaiting implementation of issues #4 & #10.

**Comment by delphidabbler** [comment]
> **Note** This proposal is dependent upon issue #7 being implemented first.

Issue #7, in turn, depends on completion of issues #4, #6 & #10!

**Comment by delphidabbler** [comment]
Decided to do this. Will have to be after next release (i.e. the release that includes compiler change).

**Comment by delphidabbler** [comment]
> > Note This proposal is dependent upon issue #7 being implemented first.
> 
> Issue #7, in turn, depends on completion of issues #4, #6 & #10!

All these issues have now been closed.

> Decided to do this. Will have to be after next release (i.e. the release that includes compiler change).

That has now been done - [release v1.1.0](https://github.com/delphidabbler/vilib/releases/tag/v1.1.0) of 2022-11-20 was built with Delphi 11.

⭐ **_So, all is ready to go!_**

**Comment by delphidabbler** [comment]
This issue will be implemented in three parts

1. Source tree will be restructured to enable extraction to new librart repo. (being done in branch `feature/8a-prepare-to-extract-core-code-to-new-repo`).
2. Code will be moved.
3. Code & repo will be adjusted to use code from new library. Need to use feature branch `feature/8b-import-core-code-from-library`

**Comment by delphidabbler** [comment]
Extracted code from `master` as of tag `v1.1.0`

Used `git filter-repo` commands

```bash
git clone --branch master https://github.com/delphidabbler/vilib.git
git filter-repo --subdirectory-filter Src/
git filter-repo --path Exports/ --invert-paths
git filter-repo --path UBinaryVerInfo.pas --path VIBinData.dpr \
  --path VIBinData.dproj --path VIBinData.cfg --path VVIBinData.vi \
  --invert-paths
```

This left just the following files in the repo root:

* `UVerInfoBinIO.pas`
* `UVerInfoData.pas`
* `UVerInfoRec.pas`

History was left intact.

The filtered repo was then pushed to `ddablib/vibin` using:

```bash
git remote add origin https://github.com/ddablib/vibin.git
git branch -M main
git push origin main
```

The point in history where the code was imported to `vibin` was tagged with the `import-from-vilib` tag:

```bash
git tag -a import-from-vilib
git push --tags
```

The `v1.0.5` & `v1.1.0` tags were replaced by `vilib-v1.0.5` and `vilib-v1.1.0` tags respectively.

**Comment by delphidabbler** [comment]
`vibin` received its 1st release, v1.0.0, on 2023-05-17.

**Comment by delphidabbler** [comment]
`vilib` revised to use code from `vibin`, imported as a submodule, instead of its own versions of that code. The code exported to `vibin` was removed from `vilib`.

Implemented at merge commit c2b0750e7c1e61fe412c3fa93df7069c19d3b833
